src/utils/resizer.ts

```diff
@@ -0,0 +1,56 @@
+import qs from "qs"
+const GEMINI_CLOUDFRONT_URL = "https://d7hftxdivxxvm.cloudfront.net"
+
+const warn = message => {
+  if (process.env.NODE_ENV === "development") {
+    console.warn(message)
+  }
+}
+
+export const crop = (src: string, options: { width: number; height: number; quality?: number }) => {
+  const { width, height, quality } = options
+
+  if (!width && !height) {
+    warn("requires width and height")
+    return src
+  } else if (width === 0) {
+    warn("width must be non-zero")
+    return src
+  } else if (height === 0) {
+    warn("height must be non-zero")
+    return src
+  }
+
+  const config = {
+    resize_to: "fill",
+    src,
+    width,
+    height,
+    quality: quality || 95,
+  }
+
+  return [GEMINI_CLOUDFRONT_URL, qs.stringify(config)].join("?")
+}
+
+export const resize = (src: string, options: { width?: number; height?: number; quality?: number }) => {
+  const { width, height, quality } = options
+
+  let resizeTo
+  if (width && !height) {
+    resizeTo = "width"
+  } else if (height && !width) {
+    resizeTo = "height"
+  } else {
+    resizeTo = "fit"
+  }
+
+  const config = {
+    resize_to: resizeTo,
+    src,
+    width,
+    height,
+    quality: quality || 95,
+  }
+
+  return [GEMINI_CLOUDFRONT_URL, qs.stringify(config)].join("?")
+}
```

Comment:
Do you double resolution for retina displays? Or is that something that’s hard to figure out when doing SSR? If not, we should really have a ticket for that and figure it out sometime.

In [Emission we do it](https://github.com/artsy/emission/blob/3a920657e68dce5a9fc9a2a61affbe9c831144ed/src/lib/Components/OpaqueImageView.tsx#L88-L96), but it’s easy because we’ll know the exact laid out size and the screen pixel density.

Reply:
We're not as precise with it -- for images that are cropped, we double exactly. But for everything else, we ask for 1200px resized width to sort of account for retina. There are definitely problems with this -- the max-width of images on the `StandardLayout` are 780px. For `FeatureLayout` they could be full-bleed. Also on mobile we use the same image size so 1200 is kind of an arbitrary in-between. This should be handled better.

For SSR in Force, we have an `isMobile` constant that is determined by the user-agent. We could use that for choosing sizes. Retina detection is a little harder with SSR. We could detect once and store it in a session cookie? 

I guess what we want here is a combo of knowing user-agent (for SSR Mobile) and some predetermined constants for various layouts. Unlike Gravity artworks, article images are just uploads to S3 (backed by Gemini/Cloudfront), and we don't have various pre-made images (i.e large, larger, medium) so either we add that kind of support in Positron (our team has reviewed this possibility a bunch and it would require quite a bit of work) or we can use predetermined constants.

Reply:
Interesting stuff to think about. We’re probably not the first ones, though, so probably should also do some research.

Re those predetermined constants; to be clear, we don’t use those in Emission, I think the future is only dynamic sizing (with appropriate caching).

Reply:
To add a me too: I had this same SSR problem in team nav, couldn't really figure out a good answer